### PR TITLE
PE-4923: Introduces methods for getting file name and extension

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: lib/src/utils/path_utils.dart [hunk 1]
```diff
@@ -88,6 +88,14 @@ String getFileExtension({
   }
 }
 
+String getFileTypeFromMime({required String contentType}) {
+  return contentType.substring(contentType.lastIndexOf('/') + 1);
+}
+
+String getBasenameWithoutExtension({required String filePath}) {
```
Comment (thiagocarvalhodev) [inline lib/src/utils/path_utils.dart:95]: @matibat , can you add the `withoutExtension` parameter and implement both options: with and without extension?

File: test/src/utils/path_utils_test.dart [hunk 2]
```diff
@@ -112,4 +112,22 @@ void main() {
       expect(ext, 'conf');
     });
   });
+
+  group('getFileTypeFromMime method', () {
+    test('should return the file type from mimetype', () {
+      final type = getFileTypeFromMime(contentType: 'application/pdf');
+
+      expect(type, 'pdf');
+    });
```
Comment (thiagocarvalhodev) [inline test/src/utils/path_utils_test.dart:121]: @matibat please add more test cases here. Chat GPT can generate for you.

File: test/src/utils/path_utils_test.dart [hunk 3]
```diff
@@ -112,4 +112,22 @@ void main() {
...
+  });
+
+  group('getBasenameWithoutExtension method', () {
+    test('should return the basename without extension', () {
+      final basename = getBasenameWithoutExtension(
+        filePath: '/hola/que/tal/file.pdf',
+      );
+
+      expect(basename, 'file');
```
Comment (thiagocarvalhodev) [inline test/src/utils/path_utils_test.dart:130]: Same comment as before. Some test cases: 
- filePath: 'file.pdf'
- filePath: 'file.pdf.txt'
- filePath: 'file' <- no extension on the file name (is it possible to get using IOFile?)

**Review by thiagocarvalhodev — CHANGES_REQUESTED** [review]
We need more test cases on the unit tests. This comment is optional: https://github.com/ar-io/ardrive_io/pull/19/files#r1385438373

File: lib/src/utils/path_utils.dart (hunk 1 above)
Comment (matibat) [reply lib/src/utils/path_utils.dart:95]: I'd prefer to introduce a new method when needed. 🙏

**Comment by matibat** [comment]
Introduced more test cases, @thiagocarvalhodev .

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM

**Pull request merged**